src/xrpld/app/misc/detail/TxQ.cpp

```diff
@@ -737,6 +737,13 @@ TxQ::apply(
     STAmountSO stAmountSO{view.rules().enabled(fixSTAmountCanonicalize)};
     NumberSO stNumberSO{view.rules().enabled(fixUniversalNumber)};
 
+    // See if the transaction is valid, properly formed,
```

Comment:
Why was this moved up?

Reply:
Oh, is this to catch the too-long `sfRawTransactions` before `calculateBaseFee` is hit?

Reply:
exactly, this move is to call preflight before the calculateBaseFee function.